src/index.js

```diff
@@ -18,7 +23,159 @@ import { months } from './data.js'
+      continentWiseObject = sortCitiesByContinent(continentWiseObject)
+      changeBottomSegment(continentWiseObject)
+    } else {
+      bottomItemCount = 0
+      continentArrow.src = '/Assets/HTML&CSS/GeneralImages&Icons/' + 'arrowDown' + '.svg'
+      continentArrow.id = 'down-arrow'
+      clearInterval(setTimerIdBottom)
+      bottomContainerElements.innerHTML = ''
+      continentWiseObject = sortCitiesByContinent(continentWiseObject).reverse()
+      changeBottomSegment(continentWiseObject)
+    }
+  }
+
+  /**
+   *  This function will  sort the cities based on continent name
+   * @param {object}  cityDetails all city details
+   * @returns  {object} sorted cities
+   */
+  function sortCitiesByContinent (cityDetails) {
+    cityDetails.sort((a, b) => {
+      return (a.timeZone.split('/')[0]) < (b.timeZone.split('/')[0]) ? -1 : 1
+    })
+    return cityDetails
+  }
+
+  /**
+   *   This function will update the bottom container elements based on temperature sorting criteria
+   * @param {string} arrowId  It is the arrow id of the arrow image
+   */
+  function changeBottomByTemperature (arrowId) {
+    const citiesTemperature = document.getElementsByClassName('temperature-arrow')[0]
+    if (arrowId === 'up-arrow-temp') {
+      bottomItemCount = 0
+      citiesTemperature.src = '/Assets/HTML&CSS/GeneralImages&Icons/' + 'arrowDown' + '.svg'
+      citiesTemperature.id = 'down-arrow-temp'
+      clearInterval(setTimerIdBottom)
+      bottomContainerElements.innerHTML = ''
+      continentWiseObject = sortCitiesByTemperature(continentWiseObject, 'desc')
+      changeBottomSegment(continentWiseObject)
+    } else {
+      bottomItemCount = 0
+      citiesTemperature.src = '/Assets/HTML&CSS/GeneralImages&Icons/' + 'arrowUp' + '.svg'
+      citiesTemperature.id = 'up-arrow-temp'
+      clearInterval(setTimerIdBottom)
+      bottomContainerElements.innerHTML = ''
+      continentWiseObject = sortCitiesByTemperature(continentWiseObject, 'asc')
+      changeBottomSegment(continentWiseObject)
+    }
+  }
 
+  /**
+   *   This function will  sort the cities based on temperature
+   * @param {object} cityDetails all city details
+   * @param {string} order order to sort the elements
+   * @returns {object}  sorted cities
+   */
+  function sortCitiesByTemperature (cityDetails, order) {
+    cityDetails.sort((a, b) => {
+      if ((a.timeZone.split('/')[0]) !== (b.timeZone.split('/')[0])) {
+        return 0
+      } else {
+        if (order === 'asc') { return parseInt(a.temperature) - parseInt(b.temperature) } else {
+          return parseInt(b.temperature) - parseInt(a.temperature)
+        }
+      }
+    })
+    return cityDetails
+  }
+
+  /**
+   *   It will add elements to the bottom container
+   * @param {object}cities all city details
+   */
+  function changeBottomSegment (cities) {
+    const bottomContainer = document.getElementsByClassName('bottom-itemscontainer')[0]
+    cities.map((city) => {
```

Comment:
If we don't need the list created from iterating over an array, we can use `forEach` or `for of` instead